hask/02.hs

```diff
@@ -20,25 +20,20 @@ main = do
     print $ solve part1 contents
     print $ solve part2 contents
 
-solve f = sum . map (line f) . zip [1..] . lines
+solve f = sum . map f . zip [1..] . lines
```

Comment:
Your `line` function basically unwraps a `Maybe`. But you only actually need it for part 1... which is returning an `Int` anyway. So you can get rid of the `Maybe` from both part 1 and 2, and just use the values directly.